data-prepper-plugins/mongodb/src/test/java/org/opensearch/dataprepper/plugins/mongo/leader/LeaderSchedulerTest.java

```diff
@@ -93,11 +97,13 @@ void test_should_init_export() throws InterruptedException {
         ExecutorService executorService = Executors.newSingleThreadExecutor();
         executorService.submit(() -> leaderScheduler.run());
         Thread.sleep(100);
-        executorService.shutdownNow();
-
 
         // Acquire the init partition
-        verify(coordinator).acquireAvailablePartition(eq(LeaderPartition.PARTITION_TYPE));
+        await()
+            .atMost(Duration.ofSeconds(2))
+            .untilAsserted(() -> verify(coordinator).acquireAvailablePartition(eq(LeaderPartition.PARTITION_TYPE)));
+
+        executorService.shutdownNow();
```

Comment:
This `shutdown` appears to end the thread. But, there may be a delay between this and when `giveUpPartition` is called.

It may make sense to add the following immediately after this line:

```
await()
            .atMost(Duration.ofSeconds(2))
            .untilAsserted(() -> verify(coordinator).giveUpPartition(any)));
```

Reply:
Updated